coal/mineral.py

```diff
@@ -44,47 +53,58 @@ def trainClassifier(libraryFileName):
     trainingData = ???(normalize(???(library)))
 
     # initialize and train a neural network using the training data
-    classifier = spectral.algorithms.classifiers.PerceptronClassifier(???)
+    classifier = spectral.classifiers.PerceptronClassifier(???)
     classifier.train(trainingData, ???)
 
-    # TODO close file ?
-
     # return trained classifier
     return classifier
 
 def saveClassifier(classifier, classifierFilename):
-
     """
-    Save a classifier to a file.
+    This saves a classifier to a file for reuse.
+    Args:
+       classifier (PerceptronClassifier): the classifier to write
+       classifierFilename (str): the file which to write the classifier
+    Returns:
+        None
     """
 
-    # TODO close file ?
-
-    pickle.dump(classifier, open(classifierFilename,"wb"))
+    with open(classifierFilename,"wb") as f:
+        pickle.dump(classifier, f)
 
 def readClassifier(classifierFilename):
-
     """
-    Read a classifier from a file.
+    This reads a classifier from a file.
+    Args:
+       classifierFilename (str): the file from which to load the classifier
+    Returns:
+        ??? not sure about how to express this
```

Comment:
my expectation was that `readClassifier` is a function from `str` to `PerceptronClassifier`, a trained classifier.